### ISSUE TITLE
Crash when appending PDF more than 507 times

### ISSUE THREAD

**Issue by shaehn** [chunyenHuang/hummusRecipe#61]
Since I had to programatically create a template file before I could edit it, I put the creation inside a loop.

```javascript
const PdfDocument  = require('hummus-recipe');

pdfDoc = new PdfDocument('new', 'junk2.pdf');
const underlay = 'PDF-document1.pdf';
for (let page = 1; page < 10001; page++) {
    console.log(page);
    pdfDoc.appendPage(underlay);
}
pdfDoc.endPDF();
```
when the loop was getting ready to append at the 508th iteration, the following appeared.

```
C:\Users\shaehn\json2pdf\node_modules\hummus-recipe\lib\appendPage.js:42
        this.writer.appendPDFPagesFromPDF(pdfSrc);
                    ^

TypeError: unable to append page, make sure it's fine
    at Recipe.appendPage (C:\Users\me\project\node_modules\hummus-recipe\lib\appendPage.js:42:21)
    at Object.<anonymous> (C:\Users\me\project\edit.js:7:16)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
```
Is there some kind of limit involved when appending an existing document to a new document?

**Comment by shaehn** [comment]
Not sure if this will give a hint as to what is going on, but if I do the .appendPage 500 times before I invoke the crash, the document created appears to be unhealthy in some way, because when I try to print out the structure of the newly created document, I get the following problem.

```
C:\Users\shaehn\json2pdf\node_modules\hummus-recipe\lib\info.js:176
            const type = inObject.getType();
                                  ^

TypeError: Cannot read property 'getType' of undefined
    at iterateObjectTypes (C:\Users\shaehn\json2pdf\node_modules\hummus-recipe\lib\info.js:176:35)
    at write (C:\Users\shaehn\json2pdf\node_modules\hummus-recipe\lib\info.js:227:9)
    at structures.forEach (C:\Users\shaehn\json2pdf\node_modules\hummus-recipe\lib\info.js:231:9)
    at Array.forEach (<anonymous>)
    at Recipe.structure (C:\Users\shaehn\json2pdf\node_modules\hummus-recipe\lib\info.js:230:16)
    at Object.<anonymous> (C:\Users\shaehn\json2pdf\edit.js:16:8)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
```

So it would appear that the appendPage is not creating healthy PDF file.

**Comment by chunyenHuang** [comment]
I will check soon. Thanks for finding the issue.

**Comment by shaehn** [comment]
fixed in release 1.9.1
